packages/plugin-credentials/src/token.js

```diff
@@ -0,0 +1,332 @@
+        }
+
+        // If the new token is the same as the previous token, then we may have
+        // found a bug in CI; log the details and reject the Promise
+        if (this.access_token === res.body.access_token) {
+          this.logger.error(`token: new token matches current token`);
+          // log the tokens if it is not production
+          if (process.env.NODE_ENV !== `production`) {
+            this.logger.error(`token: current token:`, this.access_token);
+            this.logger.error(`token: new token:`, res.body.access_token);
+          }
+          return Promise.reject(new Error(`new token matches current token`));
+        }
+
+        if (this.previousToken) {
+          this.previousToken.revoke();
+          this.unset(`previousToken`);
+        }
+
+        res.body.previousToken = this;
+        res.body.scope = this.scope;
+
+        return new Token(res.body, {parent: this.parent});
+      })
+      .catch(processGrantError)
+      .then(tap(() => {
+        this.logger.info(`token: access token refreshed`);
+        this.isRefreshing = false;
+      }))
+      .catch((res) => {
+        this.logger.warn(`token: failed to refresh access token`, res);
+        this.isRefreshing = false;
+        return Promise.reject(res);
+      });
+  },
+
+  @oneFlight
+  revoke() {
+    if (this.isExpired) {
+      this.logger.info(`token: access token already expired, not revoking`);
+      return Promise.resolve();
+    }
+
+    if (!this.canAuthorize) {
+      this.logger.info(`token: access token no longer usable, not revoking`);
+      return Promise.resolve();
+    }
+
+    this.logger.info(`token: revoking access token`);
+    return this.spark.request({
+      method: `POST`,
+      service: `oauth`,
+      resource: `/revoke`,
+      form: {
+        token: this.access_token,
+        token_type_hint: `access_token`
+      },
+      auth: {
+        user: this.config.oauth.client_id,
+        pass: this.config.oauth.client_secret,
+        sendImmediately: true
+      },
+      shouldRefreshAccessToken: false
+    })
+      .then(() => {
+        this.unset([
+          `access_token`,
+          `expires`,
+          `expires_in`,
+          `token_type`
+        ]);
+        this.logger.info(`token: access token revoked`);
+      })
+      .catch(processGrantError);
+  },
+
+  toString() {
+    if (!this.string) {
+      throw new Error(`cannot stringify Token`);
+    }
+
+    return this.string;
+  },
+
+  validate() {
+    return this.spark.request({
+      method: `POST`,
+      service: `conversation`,
+      resource: `users/validateAuthToken`,
+      body: {
+        token: this.access_token
+      }
+    })
+      .catch((reason) => {
+        if (reason.message === `\`conversation\` is not a known service`) {
+          const convApi = process.env.CONVERSATION_SERVICE || process.env.CONVERSATION_SERVICE_URL || `https://conv-a.wbx2.com/conversation/api/v1`;
```

Comment:
So if we don't seem to have conversation in the service catalog we try again with a url that's in the environment., and as a last result fallback to this hard-coded url. Should this be something in the defaults file?